frontend/src/api/actions/new_position.js

```diff
@@ -0,0 +1,47 @@
+import { error } from "react-notification-system-redux";
+import { errorProps } from "../../modules/notifications/constants";
+import {
+    CREATE_NEW_POSITION_SUCCESS,
+    IMPORT_NEW_POSITION_FAILURE,
+    IMPORT_NEW_POSITION_SUCCESS
+} from "../constants/new_position";
+import { apiPOST } from "../../libs/apiUtils";
+
+export const createNewPosition = payload => async dispatch => {
+    try {
+        const data = await apiPOST("/positions", payload);
+        dispatch(createNewPositionSuccess(data));
+        window.location = "/tapp/positions";
```

Comment:
Never, never, never set `window.location` yourself. API actions should be side-effect free. That is, they can change the model state, but nothing else (no changing of routes, no changing of UI state)

Reply:
When you dispatch something in the UI, you can `await` the dispatch. That is the appropriate place to change the route.

Reply:
Moved route changing to view